internal/functions/gerritstatusupdater/gerritstatusupdater.go

```diff
@@ -147,30 +148,92 @@ const (
-	// instead the messages that would be sent to GerritHub are written to
-	// stdout.
-	EnvDebug = EnvPrefix + "DEBUG"
+	// EnvDryRun can be set to a non-empty string to prevent any actual "writes"
+	// happening to either GitHub and/or GerritHub.
+	EnvDryRun = EnvPrefix + "DRYRUN"
 )
 
 const (
-	// GitHub does not define constants for status and conclusion fields
-	// on workflow jobs and runs. Define some here.
+	// GitHub does not define constants for event action, status and conclusion
+	// fields on workflow jobs and runs. Define some here.
+	//
+	// The status and conclusion values are shared between workflow runs and
+	// jobs. This is not officially documented at
+	// https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads,
+	// but it's demonstrably so in practice... which is all we've got.
+
+	eventActionRequested  = "requested"
+	eventActionInProgress = "in_progress"
+	eventActionCompleted  = "completed"
+
+	workflowStatusInProgress = "in_progress"
+	workflowStatusCompleted  = "completed"
 
-	workflowStatusQueued      = "queued"
-	workflowStatusCompleted   = "completed"
 	workflowConclusionSuccess = "success"
 )
 
 // ServeHTTP is the implementation of the gerritstatusupdater serverless
 // function.
 func (fn Function) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	// c (context) represents the state we build up as we receive and handle
+	// events. We capture this as a struct to simply the logging of context as
+	// we progressively populate the state.
+	//
+	// Do not put anything sensitive in this struct. It might be logged.
+	var c struct {
+		// DeliveryID is the ID associated with the webhook event
+		// sent by GitHub.
+		DeliveryID string
+
+		// Repo is "{owner}/{Repo}" (not any URL, and not just the Repo name).
+		Repo string `json:",omitempty"`
+
+		// WorkflowRunID is the ID of the workflow run for which we either
+		// receive an event directly or via one of its associated workflow jobs.
+		WorkflowRunID *int64 `json:",omitempty"`
+
+		// WorkflowPath is the path to yaml file that defines the workflow
+		// declaration behind the workflow run and its jobs.
+		WorkflowPath string `json:",omitempty"`
+
+		// HeadBranch is the branch for which a workflow is triggered.
+		HeadBranch string `json:",omitempty"`
+	}
+
+	var (
+		// workflowName is the name of the workflow run behind the event
+		workflowName string
+
+		// msg is the human message on the Gerrit CL review
+		msg string
+
+		// result is set on the Gerrit CL review as a score.
+		result *int
+	)
+
+	c.DeliveryID = github.DeliveryID(r)
+
+	// Set up simple logging that closes over certain variables
+	if os.Getenv("NETLIFY") == "true" {
+		logpkg.SetFlags(0) // we get time information from Netlify
+	}
+	// We also get a uniqure prefix from Netlify, but build
+	// a context-based prefix ourselves here too.
+	logpkg.SetPrefix("")
+	log := func(format string, args ...interface{}) {
```

Comment:
nit: I would call this `logf`, then you don't need to rename `log` either.

Reply:
Done.